Source/ACE.Server/WorldObjects/Managers/EmoteManager.cs

```diff
@@ -1367,30 +1397,35 @@ public float ExecuteEmote(BiotaPropertiesEmote emoteSet, BiotaPropertiesEmoteAct
         /// </summary>
         public BiotaPropertiesEmote GetEmoteSet(EmoteCategory category, string questName = null, VendorType? vendorType = null, uint? wcid = null, bool useRNG = true)
         {
-            // always pull emoteSet from _worldObject
-            var emoteSet = _worldObject.Biota.BiotaPropertiesEmote.Where(e => e.Category == (uint)category);
+            var emoteSet = new List<BiotaPropertiesEmote>();
+            var hasValidEmoteSets = false;
+            if (vendorType != null)
+                hasValidEmoteSets = emoteCacheVendor.TryGetValue(vendorType.Value, out emoteSet);
+            else
+                hasValidEmoteSets = emoteCache.TryGetValue(category, out emoteSet);
+
+            if (!hasValidEmoteSets)
+                return null;
 
             // optional criteria
             if (questName != null)
-                emoteSet = emoteSet.Where(e => e.Quest.Equals(questName, StringComparison.OrdinalIgnoreCase));
-            if (vendorType != null)
-                emoteSet = emoteSet.Where(e => e.VendorType != null && e.VendorType.Value == (uint)vendorType);
+                emoteSet = emoteSet.Where(e => e.Quest.Equals(questName, StringComparison.OrdinalIgnoreCase)).ToList();
```

Comment:
this shoudn't be casting .ToList() every single clause

this is effectively turning 1 query into 4-5 separate queries